BenchPress/Generators/ResourceTypes/KeyVaultKey.cs

```diff
@@ -0,0 +1,22 @@
+namespace Generators.ResourceTypes;
+
+public class KeyVaultKey : ResourceType
+{
+    public KeyVaultKey() { }
+
+    public override string Id => "Microsoft.KeyVault/vaults/keys";
+    public override string FullName => Id;
+    public override string FriendlyName => "Key Vault Key";
+    public override string Prefix => "kvk";
+    public override string FunctionPrefix => "KeyVaultKey";
+
+    public override IEnumerable<KeyValuePair<string, object>> GetResourceParameters(TestMetadata m)
+    {
+        return new[]
+        {
+            Param("ResourceType", "KeyVaultKey"),
+            Param("ResourceName", m.ResourceName),
+            Param("KeyVaultName", m.ExtraProperties["name"])
```

Comment:
Similar comment. Should the key be `vaults`?

Reply:
Maybe I misunderstood our conversation the other day, but all of the `ExtraProperties` keys come from the generated ARM template, no?

Reply:
Example:

``` JSON
    {
      "type": "Microsoft.KeyVault/vaults/keys",
      "apiVersion": "2022-07-01",
      "name": "[format('{0}/{1}', parameters('name'), 'samplekey')]",
      "properties": {
        "kty": "RSA",
        "keyOps": [],
        "keySize": 2048,
        "curveName": "P-521"
      },
      "dependsOn": [
        "[resourceId('Microsoft.KeyVault/vaults', parameters('name'))]"
      ]
    }

```

Reply:
Yes, it does come from the ARM template. I explained a bit how the extraProperties dictionary is created in the comments of the method: https://github.com/Azure/benchpress/blob/1f759817506f678202ee7d5300ef7c8d6502c281/BenchPress/Generators/AzureDeploymentImporter.cs#L145

For your example, the `extraProperties` dictionary would look like the following:
```
{
    "vaults": "parameters('name'))]" <- This reminds me that I need to create an issue to correctly strip any trailing braces
}
```

For something like ApiManagementPolicy, I assume the `dependsOn` object looks like this:
```
"[resourceId('Microsoft.ApiManagement/service/apis', parameters('servicename'), parameters('apiname'))]"
```
which would result in the following dictionary:
```
{
    "service": parameters('servicename')
    "apis": parameters('apiname'))]
}
```

In short, the resource type is used to create the keys, and then the parameter names are matched with the respective keys.

Reply:
So, did I do it right? LOL 

The generated ARM is:

``` JSON
    {
      "type": "Microsoft.ApiManagement/service/apis/policies",
      "apiVersion": "2022-08-01",
      "name": "[format('{0}/{1}/{2}', parameters('serviceName'), parameters('apiName'), parameters('policyName'))]",
      "properties": {
        "format": "rawxml",
        "value": "[variables('$fxv#0')]"
      },
      "dependsOn": [
        "[resourceId('Microsoft.ApiManagement/service/apis', parameters('serviceName'), parameters('apiName'))]"
      ]
    }
```

Reply:
No, you did not.

Again, if that is the generated ARM for API Policies, the following key/value pairs will be set in the `extraProperties` dictionary:
```
{
    "service": parameters('serviceName')
    "apis": parameters('apiName'))]
}
```
Then in the `ApiManagementApi` class, the following `Param`s can be initialized:
```
Param("ServiceName", m.ExtraProperties["service"]),
Param("ApiId", m.ExtraProperties["apis"])
```
However, if there is still confusion, I would prefer to understand anything else I can clear up. Or we can go over in design session, so that it's clear how to introduce new ResourceType classes.

Reply:
AH!!! I GET IT!!!

I don't know why it took me so long to wrap my head around it, but now it makes sense.

Update coming up!

Reply:
Yay, awesome! Apologies if I introduced any confusion. If you feel like more descriptive comments could be added to this method, https://github.com/Azure/benchpress/blob/1f759817506f678202ee7d5300ef7c8d6502c281/BenchPress/Generators/AzureDeploymentImporter.cs#L167
I am more than happy to add some better documentation and examples! It made so much sense to me during implementation. Looking back though, it's definitely super confusing LOL  🤣

Reply:
Where this really threw me for a loop is when there is more than one element in the path (e.g., `Microsoft.ApiManagement/service/apis`) and how that actually created two keys `service` and `apis` and how the parameters map to those two keys. Essentially, what is the relationship between the path, the keys, and the values.

Now that I understand that, it's pretty easy.

The only thing that still kind of makes me think twice is that I wonder if this is a universal truth. Meaning will this relationship between the path, the keys, and the values always be true (e.g., will there ever be a /service/someservice has three parameters). So far, I haven't found on, so I assume that it's a built in universal truth.

Reply:
I see in the code that if the path's count doesn't match the parameters count it doesn't do anything! LOL